src/scala/org/pantsbuild/zinc/compiler/Main.scala

```diff
@@ -73,49 +64,77 @@ object Main {
     cl
   }
 
+  def preprocessArgs(args: Array[String]): Array[String] = {
+    val fixedArgs = args.flatMap { arg =>
+      arg match {
+        case x if x.startsWith("-C") || x.startsWith("-S") => {
+          val tup = arg.splitAt(2)
+          Seq(tup._1, tup._2)
+        }
+        case arg => Seq(arg)
+      }
+    }
+    for (i <- 1 to fixedArgs.size) {
+      // Because we set -cp as the shorthand for classpath, we can't also set -classpath as one.
+      // Fix it up.
+      if (fixedArgs(i - 1) == "-classpath") {
+        fixedArgs(i - 1) = "--classpath"
+      }
+      // Old versions of this binary used :s not ,s as list separators.
+      // Accept their input.
```

Comment:
Backwards compatibility is not a strict concern here... 99% of consumers use the built-in zinc version, and Twitter only overrides the version to add some debugging tools to a classpath that matches the built-in version.